builds/e2e/upgrade-scenarios.yaml

```diff
@@ -0,0 +1,204 @@
+trigger: none
+pr: none
+
+resources:
+  pipelines:
+  - pipeline: images
+    source: 'Azure-IoT-Edge-Core Build Images'
+    branch: 'release/1.2'
+    trigger:
+      branches:
+      - release/1.2
+  - pipeline: packages
+    source: 'Azure-IoT-Edge-Core Edgelet Packages'
+    branch: 'release/1.2'
+    trigger:
+      branches:
+      - release/1.2
+  - pipeline: images_main
+    source: 'Azure-IoT-Edge-Core Build Images'
+    branch: 'main'
+  - pipeline: images_release_1_1
+    source: 'Azure-IoT-Edge-Core Build Images'
+    branch: 'release/1.1'
+
+jobs:
+################################################################################
+  - job: partial_upgrade_ubuntu_1804_msmoby
+################################################################################
+    displayName: Partial Upgrade
+
+    pool:
+      name: $(pool.linux.name)
+      demands:
+      - ImageOverride -equals agent-aziotedge-ubuntu-18.04-msmoby
+
+    variables:
+      os: linux
+      arch: amd64
+      artifactName: iotedged-ubuntu18.04-amd64
+      identityServiceArtifactName: packages_ubuntu-18.04_amd64
+      identityServicePackageFilter: aziot-identity-service_*_amd64.deb
+      minimal: true
+      verbose: false
+
+    strategy:
+      matrix:
+        1.2_to_1.3_no_edge_daemon:
+          edgeDaemonVersion: "release/1.2"
+          bootstrapAgentVersion: "main"
+          runtimeModulesVersion: "main"
+        1.2_to_1.3_only_runtime_modules:
+          edgeDaemonVersion: "release/1.2"
+          bootstrapAgentVersion: "release/1.2"
+          runtimeModulesVersion: "main"
+        1.2.0_to_1.3_no_edge_daemon:
+          edgeDaemonVersion: "1.2.0"
+          bootstrapAgentVersion: "main"
+          runtimeModulesVersion: "main"
+        1.2.0_to_1.3_only_runtime_modules:
+          edgeDaemonVersion: "1.2.0"
+          bootstrapAgentVersion: "1.2.0"
+          runtimeModulesVersion: "main"
+        1.2.5_to_1.3_no_edge_daemon:
+          edgeDaemonVersion: "1.2.5"
+          bootstrapAgentVersion: "main"
+          runtimeModulesVersion: "main"
+        1.2.5_to_1.3_only_runtime_modules:
+          edgeDaemonVersion: "1.2.5"
+          bootstrapAgentVersion: "1.2.5"
+          runtimeModulesVersion: "main"
+        1.2.9_to_1.3_no_edge_daemon:
+          edgeDaemonVersion: "1.2.9"
+          bootstrapAgentVersion: "main"
+          runtimeModulesVersion: "main"
+        1.2.9_to_1.3_only_runtime_modules:
+          edgeDaemonVersion: "1.2.9"
+          bootstrapAgentVersion: "1.2.9"
+          runtimeModulesVersion: "main"
+
+    timeoutInMinutes: 20
```

Comment:
The default value for timeout is 60 minutes. I'd only bother setting this if you need more time, not less.